backend/webserver/views.py

```diff
@@ -69,6 +72,112 @@ def post(self, request):
         logout(request)
         return Response({'message': 'Successfully Logged out'}, status=status.HTTP_200_OK)
 
+class PostView(APIView):
+    authentication_classes = [BasicAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+    def get_post(self,post_id,author_id):
+        post = get_object_or_404(Post,id=post_id,author=author_id)
+        return post
+    
+    def get(self,request,pk,post_id, *args, **kwargs):
+        author =self.get_author(pk=pk)
+        post = self.get_post(post_id,author.id)
+        if post.visibility == "PUBLIC":
+            serializer = PostSerializer(post, context={'request': request})
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            raise Http404
```

Comment:
```suggestion
            return Response({'message': 'requested post is not public'}, status=status.HTTP_400_BAD_REQUEST)
```
This is an incorrect use of the 404 status code. 404 means that something is not found which incorrect because a post was found, it was just not public. You should return a more meaningful message to the user.